### ISSUE TITLE
Exception for DataStore Hub Event subscriptionDataProcessed

### ISSUE THREAD

**Issue by raywhiteside** [aws-amplify/amplify-flutter#287]
**Describe the bug**
When using a basic DataStore setup and listening to the DataStore event stream, this exception is all over the logs:

```
I/flutter (21978): PlatformException(AmplifyException, There was a problem parsing a DataStore Hub Event.,
{PLATFORM_EXCEPTIONS: {platform: Android, localizedErrorMessage: There was a problem parsing the subscriptionDataProcessed Hub
Event., recoverySuggestion: See logs for details.}}, null)
```

Many other events seem to be parsed just fine, this is the only one I see this error for.

I'm not sure which logs this is referring to look in for further details, if you point me to those logs I can provide them.

**To Reproduce**
This exception is thrown every time a model subscription update fires when listening to the listenToDataStore stream.

**Platform**
Amplify Flutter current supports iOS and Android.  This issue is reproducible in (check all that apply):
[x] Android
[] iOS

**Comment by kjones** [comment]
What version of the Flutter SDK are you using?

I think `0.0.1-dev5` may include a fix for this from [this Andriod 1.6.8 SDK PR](https://github.com/aws-amplify/amplify-android/pull/1052).

**Comment by raywhiteside** [comment]
Ah I'm on dev4, I'll try dev5 and update in the next day or two.

**Comment by jamesonwilliams** [comment]
Hi @kjones -- your PR updated the [`OutboxMutationProcessed` event](https://docs.amplify.aws/lib/datastore/datastore-events/q/platform/android#outboxmutationprocessed).

`SubscriptionDataProcessed` is an undocumented event that is published by Android but not iOS. The Flutter project should not be logging a warning when it sees this event type, since it is an expected system behavior. The Flutter project could swallow the event and/or log a verbose log, since the event won't have an iOS equivalent.

Customers can safely ignore the message until the Flutter code is updated.

**Comment by kjones** [comment]
Ah yes. I didn't read @raywhiteside's description close enough. I was assuming the exception was causing an app crash.

**Comment by haverchuck** [comment]
@raywhiteside @kjones We will be changing how these Hub events work in a subsequent release, so that they are actually instantiated into dart types with typed payloads when applicable.  As part of this, we will be less intrusive with our error handling and will not throw errors in this manner.

**Comment by raywhiteside** [comment]
mmm sorry it is still spitting out this:
```
E/amplify:flutter:datastore( 7747): Unrecognized DataStoreHubEvent: subscriptionDataProcessed
```

**Comment by HuiSF** [comment]
@raywhiteside thanks for the follow up, `subscriptionDataProcessed` is just a normal event, the amplify-flutter DataStore Kotlin categorized it as unrecognized as the DataStore doesn't need to respond to it. However, the implement wrong used `Log.e` method which output this log as exception. Create a PR to fix this for a better developer experience.

**Comment by raywhiteside** [comment]
Perhaps instead of demoting the error to an info, catch all the known events that don't need to be responded to and print info for those? That way if there is truly an unrecognized data type it still errors, which seems preferable. 

In any case this fix will certainly clean up my logs, thanks!

**Comment by HuiSF** [comment]
Sorry @raywhiteside I meant that I've created a PR to correct this behavior.

**Comment by HuiSF** [comment]
Logging improvements on unhandled event has been released with version [0.2.3](https://github.com/aws-amplify/amplify-flutter/blob/main/packages/amplify_datastore/CHANGELOG.md#023-2021-09-09).
